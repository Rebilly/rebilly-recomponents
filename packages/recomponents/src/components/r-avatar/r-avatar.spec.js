import {shallowMount, mount, createLocalVue} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RAvatar from './r-avatar.vue';
import RIcon from '../r-icon/r-icon.vue';
import RImg from '../r-img/r-img.vue';
import RFsBlock from '../../directives/r-fs-block';

global.console.warn = jest.fn();

const localVue = createLocalVue();

localVue.directive('fb-block', RFsBlock);

describe('r-avatar.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RAvatar, {
            localVue,
            propsData: {
                initials: 'RB',
            },
            slots: {
                default: 'Avatar component',
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = await renderToString(RAvatar, {
            localVue,
            propsData: {
                initials: 'VP',
            },
            slots: {
                default: 'Avatar component',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render icon', async () => {
        const wrapper = mount(RAvatar, {
            localVue,
            propsData: {
                icon: 'heart',
            },
            stubs: {
                'r-icon': RIcon,
            },
        });

        expect(wrapper.find('.r-avatar-icon').exists()).toBeTruthy();
        expect(wrapper.find(RIcon).exists()).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render image', async () => {
        const wrapper = mount(RAvatar, {
            localVue,
            propsData: {
                image: 'https://www.meme-arsenal.com/memes/86b1de6ff6981be012d40bf26c6992ab.jpg',
            },
            stubs: {
                'r-img': RImg,
            },
        });

        expect(wrapper.find('.r-avatar-image').exists()).toBeTruthy();
        expect(wrapper.find(RImg).exists()).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger validation warning', async () => {
        const wrapper = shallowMount(RAvatar, {
            localVue,
            propsData: {
                initials: 'Rebilly',
                color: 'green',
            },
        });

        expect(global.console.warn).toBeCalled();
        expect(wrapper).toMatchSnapshot();
    });
});
