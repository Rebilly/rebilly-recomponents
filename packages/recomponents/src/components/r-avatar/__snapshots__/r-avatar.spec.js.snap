// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-avatar.vue should render Wrapper and match snapshot 1`] = `
<div class="r-avatar r-avatar-description">
  <div class="r-avatar-main fs-block">
    <div class="r-avatar-initials r-avatar-initials-accent"><span>RB</span></div>
    <!---->
    <!---->
  </div>
  <div class="r-avatar-info">
    <!----> Avatar component</div>
</div>
`;

exports[`r-avatar.vue should render icon 1`] = `
<div class="r-avatar r-avatar-description">
  <div class="r-avatar-main fs-block">
    <!---->
    <div class="r-avatar-icon"><svg class="r-icon r-icon-20">
        <use xlink:href="#icon-heart" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
      </svg></div>
    <!---->
  </div>
  <div class="r-avatar-info">
    <!---->
  </div>
</div>
`;

exports[`r-avatar.vue should render image 1`] = `
<div class="r-avatar r-avatar-description">
  <div class="r-avatar-main fs-block">
    <!---->
    <!---->
    <div class="r-avatar-image">
      <figure role="img" class="r-component r-img" style="min-height: 0px; min-width: 0px;"><img src="https://www.meme-arsenal.com/memes/86b1de6ff6981be012d40bf26c6992ab.jpg" alt="Recomponents image"></figure>
    </div>
  </div>
  <div class="r-avatar-info">
    <!---->
  </div>
</div>
`;

exports[`r-avatar.vue should render via SSR and match snapshot 1`] = `
<div class="r-avatar r-avatar-description">
  <div class="r-avatar-main">
    <div class="r-avatar-initials r-avatar-initials-yellow"><span>VP</span></div>
    <!---->
    <!---->
  </div>
  <div class="r-avatar-info">
    <!----> Avatar component</div>
</div>
`;

exports[`r-avatar.vue should trigger validation warning 1`] = `
<div class="r-avatar r-avatar-description">
  <div class="r-avatar-main fs-block">
    <div class="r-avatar-initials r-avatar-initials-green"><span>Rebilly</span></div>
    <!---->
    <!---->
  </div>
  <div class="r-avatar-info">
    <!---->
  </div>
</div>
`;
